### PR TITLE
Memex 3427 add scrapinghub

### PR DESCRIPTION
**Pull request by marsenau**
Add scrapingJob class and spider class for use in scraping alerts.

### CODE REVIEW

File: src/main/java/software/uncharted/tellfinder/data/model/ScrapingJob.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * Key that points to the spider URL
+     */
+    private String key;
+
+
+
+    public ScrapingJob(JsonNode job) {
+        this.setErrors(0);
+        this.setItems(0);
+        for (Iterator<String> fields = job.fieldNames(); fields.hasNext();) {
```
Comment (chris-dickson) [inline src/main/java/software/uncharted/tellfinder/data/model/ScrapingJob.java]: This shouldn’t be a for loop. Just set the fields directly based on what you know. 

setSpider(job.at(“/spider”))

Adding the loop doesn’t do anything.

File: src/main/java/software/uncharted/tellfinder/data/model/Spider.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                .map(job -> {
+                    return job.getItems();
+                })
+                .collect(Collectors.toList());
+        Collections.reverse(itemCount);
+
+        Map<Object, Long> startTimes = jobs.stream()
+                .map(job -> {
+                Date date = new Date(Long.parseLong(job.getStartTime().toString()));
+                    Calendar cal = Calendar.getInstance();
```
Comment (chris-dickson) [inline src/main/java/software/uncharted/tellfinder/data/model/Spider.java]: Calendar is being phased out of java. Use the Date class.

File: src/main/java/software/uncharted/tellfinder/data/model/Spider.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        this.setErrorDev(errorRollingSD.get(errorRollingSD.size()-1));
+
+        Integer lastJobItemCount = this.getJobs().get(0).getItems();
+        Integer lastJobErrorCount = this.getJobs().get(0).getErrors();
+        this.setItemZScore((lastJobItemCount-this.getItemAvg()-this.getItemSeasonComponent())/(1.0*this.getItemDev()));
+        this.setErrorZScore((lastJobErrorCount-this.getErrorAvg())/(1.0*this.getErrorDev()));
+    }
+
+    public String detectAnomalies() {
+        ScrapingJob lastJob = this.getJobs().get(0);
```
Comment (chris-dickson) [inline src/main/java/software/uncharted/tellfinder/data/model/Spider.java]: Don’t build HTML in here. Create a mode that describes the anomalies, and return it to the client.

**Comment by marsenau** [comment]
This should be in tf-common

**Pull request closed without merging**